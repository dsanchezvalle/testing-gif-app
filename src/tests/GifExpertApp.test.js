import React from 'react';
import {shallow} from 'enzyme';
import {GifExpertApp} from '../GifExpertApp'

describe('GifExpertApp tests', ()=>{
    test('it should render GifExpertApp properly', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('it should render a list of categories', () => {
        const categories = ['Goku', 'Pokemon'];
        console.log(categories.length)
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        //console.log(wrapper.find('GifGrid').debug());
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})