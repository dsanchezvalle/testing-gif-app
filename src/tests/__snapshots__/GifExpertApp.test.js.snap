// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests it should render GifExpertApp properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp tests it should render a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
  </ol>
</Fragment>
`;
